File: modules/jooby-apt/src/test/java/tests/i3705/C3705_.java
/*
 * Jooby https://jooby.io
 * Apache License Version 2.0 https://jooby.io/LICENSE.txt
 * Copyright 2014 Edgar Espina
 */
package tests.i3705;

@io.jooby.annotation.Generated(C3705.class)
public class C3705_ implements io.jooby.MvcExtension {
  protected java.util.function.Function<io.jooby.Context, C3705> factory;

  public C3705_() {
    this(new C3705());
  }

  public C3705_(C3705 instance) {
    setup(ctx -> instance);
  }

  public C3705_(Class<C3705> type) {
    setup(ctx -> ctx.require(type));
  }

  public C3705_(java.util.function.Supplier<C3705> provider) {
    setup(ctx -> provider.get());
  }

  public C3705_(java.util.function.Function<Class<C3705>, C3705> factory) {
    setup(ctx -> factory.apply(C3705.class));
  }

  private void setup(java.util.function.Function<io.jooby.Context, C3705> factory) {
    this.factory = factory;
  }

  public void install(io.jooby.Jooby app) throws Exception {}
}

File: modules/jooby-openapi/src/test/kotlin/kt/i3705/C3705.kt
/*
 * Jooby https://jooby.io
 * Apache License Version 2.0 https://jooby.io/LICENSE.txt
 * Copyright 2014 Edgar Espina
 */
package kt.i3705

import io.jooby.Jooby
import io.jooby.MvcExtension
import io.jooby.SneakyThrows
import io.jooby.annotation.GET
import io.jooby.annotation.Path
import io.jooby.kt.Kooby

@Path("/")
class C3705 {
  @GET("/search")
  fun search(): String {
    return "Hello"
  }
}

class BeanScope {
  fun <T> get(type: Class<T>): T {
    return type.newInstance()
  }
}

@io.jooby.annotation.Generated(C3705::class)
class C3705_ : MvcExtension {

  constructor() {}

  constructor(provider: SneakyThrows.Function<Class<C3705>, C3705>) {}

  override fun install(application: Jooby) {
    TODO("Not yet implemented")
  }
}

class App3705 : Kooby({ mvc(C3705_()) })

class App3705b :
  Kooby({
    val beanScope = BeanScope()
    mvc(C3705_(beanScope::get))
  })
